Mathlib/Algebra/Group/Pointwise/Set/Card.lean

```diff
@@ -3,60 +3,88 @@ Copyright (c) 2024 Yaël Dillies. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yaël Dillies
 -/
-import Mathlib.Algebra.Group.Pointwise.Finset.Basic
+import Mathlib.Data.Set.Pointwise.Finite
 import Mathlib.SetTheory.Cardinal.Finite
 
 /-!
-# Cardinalities of pointwise operations on sets.
+# Cardinalities of pointwise operations on sets
 -/
 
-namespace Set
-
-open Pointwise
-
-variable {α β : Type*}
+open scoped Cardinal Pointwise
 
-section MulAction
-variable [Group α] [MulAction α β]
+namespace Set
+variable {G M α : Type*}
 
-@[to_additive (attr := simp)]
-lemma card_smul_set (a : α) (s : Set β) : Nat.card ↥(a • s) = Nat.card s :=
-  Nat.card_image_of_injective (MulAction.injective a) _
+section Mul
+variable [Mul M] {s t : Set M}
 
-end MulAction
+@[to_additive]
+lemma cardinalMk_mul_le : #(s * t) ≤ #s * #t := by rw [← image2_mul]; exact Cardinal.mk_image2_le
 
-section IsCancelMul
-variable [Mul α] [IsCancelMul α] {s t : Set α}
+variable [IsCancelMul M]
 
 @[to_additive]
-lemma card_mul_le : Nat.card (s * t) ≤ Nat.card s * Nat.card t := by
+lemma natCard_mul_le : Nat.card (s * t) ≤ Nat.card s * Nat.card t := by
   classical
   obtain h | h := (s * t).infinite_or_finite
   · simp [Set.Infinite.card_eq_zero h]
   obtain ⟨hs, ht⟩ | rfl | rfl := finite_mul.1 h
-  · lift s to Finset α using hs
-    lift t to Finset α using ht
-    rw [← Finset.coe_mul]
-    simpa [-Finset.coe_mul] using Finset.card_mul_le
+  · unfold Nat.card
+    rw [← Cardinal.toNat_mul]
+    gcongr
+    · exact Cardinal.mul_lt_aleph0 hs.lt_aleph0 ht.lt_aleph0
+    · exact cardinalMk_mul_le
   all_goals simp
 
-end IsCancelMul
+@[to_additive (attr := deprecated (since := "2024-09-30"))] alias card_mul_le := cardinalMk_mul_le
+
+end Mul
 
 section InvolutiveInv
-variable [InvolutiveInv α] {s t : Set α}
+variable [InvolutiveInv G] {s t : Set G}
+
+@[to_additive (attr := simp)]
+lemma cardinalMk_inv (s : Set G) : #↥(s⁻¹) = #s := by
```

Comment:
`cardinalMk` is not the correct name for these things. Please use `mk`, which is the mathlib convention.

Reply:
You can get yourself acquainted with the mathlib naming convention here: https://leanprover-community.github.io/contribute/naming.html#identifiers-and-theorem-names

This passage in particular is relevant:
> When referring to a namespaced definition in a lemma name not in the same namespace, the definition should have its namespace removed. If the definition name is unambiguous without its namespace, it can be used as is. Else, the namespace is prepended back to it in `lowerCamelCase`. This is to ensure that `_`-separated strings in a lemma name correspond to a definition name or connective.

Reply:
Indeed, that passage is relevant, and it indicates that this lemma should be in the cardinal namespace.

Reply:
What? It's already in the `Set` namespace since it's primarily about sets.

Reply:
Note that almost all lemmas involving the cardinality of sets are in the Cardinal namespace, not the Set namespace. As such, this one should be as well.